sourcegraph/sourcegraph
# source: fix text overflow when displaying org name in user settings sidebar

[Slack thread](https://sourcegraph.slack.com/archives/C03K05FCRFH/p1692701165077759)

Egg-shaped org avatars were discovered earlier today; I dug into it to find out why. I discovered that it happened when the organization's name was too long to display in the side bar. The text pushed the avatar until it had no more space, which caused it to be egg-shaped.

| Before  | After  |
|---|---|
| ![image](https://github.com/sourcegraph/sourcegraph/assets/25608335/d891a738-cd73-4479-ac11-5dac487ab9ea)  |  ![CleanShot 2023-08-22 at 12 45 01](https://github.com/sourcegraph/sourcegraph/assets/25608335/0d11dd85-2e4f-4482-ba53-a709852690b7) |

This PR adds a text-overflow to the text and also adds a tooltip - this ensures that long org names are truncated and the tooltip will always display the name of the org in full.

## Test plan



* Create an org with a long name.
<img width="701" alt="CleanShot 2023-08-22 at 12 49 44@2x" src="https://github.com/sourcegraph/sourcegraph/assets/25608335/90b9596e-a61c-4c8c-9c8b-05059e660bf6">
* Navigate to the user settings page `/users/{username}/settings`
* The organization name should be truncated and the avatar shouldn't be egg-shaped.
